### PR TITLE
Rename oe-gdb to oegdb and add it as a target for the CMake package

### LINKED ISSUE #1599: Need a cmake openenclave::oe-gdb SDK target
To sign an enclave within cmake you can use openenclave::oesign for the binary name and it will pull it from the SDK bin directory.
To add a custom target to run you generally need to have a custom command to run `host/host enc/enclave`. No special SDK target is needed here.
However we should be able to add a custom target so I can debug. This way I should be able to add a custom target `debug` which calls openenclave::oe-gdb, but this does not exist.
The reason the lack of this became problematic for me is that I need to run the host as sudo and that loses the path so it cannot find it so I had to jump through hoops to get the path before I call sudo,
Adding an openenclave::oe-gdb would make this easier.

### PR DESCRIPTION
**Pull request by andschwa**
Resolves #1599. It was a little more complicated than using `install` because it's not a target in our build.

### CODE REVIEW

**Comment by anitagov** [comment]
To add to @anakrish 's concern, why is this breaking change getting into Release 0.5?

**Comment by andschwa** [comment]
Wouldn’t it be the best time to do it? We can skip the rename if you want, but that just punts it to v0.6.0. I would reason that the earlier we make these breaking changes, the better, so I would push for v0.5.0.

**Comment by anitagov** [comment]
@andschwa - OK. @anakrish will need to make the call on the renaming.

**Comment by andschwa** [comment]
> something does not look to be working though. My `make debug` results in `openenclave::oegdb: command not found`

This was because `COMMAND` only replaces the target with the location of the executable if it is the first argument, but in your sample `sudo` was the first argument. I confirmed with @paulcallen offline that chaning to `COMMAND openenenclave::oegdb` (and then running `sudo make debug`) worked as expected, and changing it to `COMMAND sudo $<TARGET_FILE:openenclave::oegdb>` also worked as expected.

**Comment by andschwa** [comment]
Whoops, this should be in v0.6.0, thanks for the catch @anakrish.

**Comment by andschwa** [comment]
This is blocked on #1613 as we need the changelog updated so I can put this entry under v0.6.0.

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
```
        Start   6: oegdb-test

  6/171 Test   #6: oegdb-test ...............................................................................................................................................   Passed    0.38 sec

        Start   7: oegdb-test-simulation-mode

  7/171 Test   #7: oegdb-test-simulation-mode ...............................................................................................................................   Passed    0.24 sec
```

**Comment by andschwa** [comment]
> Alternative suggestion: we could if people really wanted go the other route and rename everything to be `oe-gdb`/`oe-sign`/`oe-edger8r` etc.

**Comment by andschwa** [comment]
I've added a script with the original name that emits a loud warning and then replaces itself with the renamed script.

bors try

File: debugger/oegdb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Get path of the oe-gdb script
+# See https://mywiki.wooledge.org/BashFAQ/028 for complexities involved
+# in determining location of a bash script. ${BASH_SOURCE}, though not perfect,
+# is an acceptable solution for oe-gdb.
```
Comment (anakrish) [inline debugger/oegdb]: Change to oegdb in comment

File: debugger/oegdb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+OE_GDB_DIR=$(dirname "${BASH_SOURCE[0]}")
+
+# Get the path to the debugger libraries relative to the oe-gdb path.
```
Comment (anakrish) [inline debugger/oegdb]: Change to oegdb in comment.

File: debugger/oe-gdb [hunk 3]
```diff
@@ -3,18 +3,10 @@
...
-# Get path of the oe-gdb script
-# See https://mywiki.wooledge.org/BashFAQ/028 for complexities involved
-# in determining location of a bash script. ${BASH_SOURCE}, though not perfect,
-# is an acceptable solution for oe-gdb.
-OE_GDB_DIR=$(dirname "${BASH_SOURCE[0]}")
-
-# Get the path to the debugger libraries relative to the oe-gdb path.
-# Normalize the path by cd-ing and doing a pwd -P.
-OE_GDB_LIB_DIR=$(cd "$OE_GDB_DIR/../lib/openenclave/debugger" || exit; pwd -P)
+echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
```
Comment (anakrish) [inline debugger/oe-gdb]: Great that build/output/oe-gdb cannot be executed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [Unreleased]
 ------------
 
+- `oe-gdb` renamed to `oegdb` for consistency with other tools, such as
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Please put this under the `### Changes` header.

Also, we never defined a consistent writing style for the changelog. I've been going by the commit message guidelines (Lead with active imperative verb), we should probably clarify that.

File: debugger/oe-gdb (hunk 3 above)
Comment (andschwa) [reply debugger/oe-gdb]: That's actually an oversight on my part. The installation step took care of the permissions for me, and I did not check just the build output. Let's leave it as is?

Comment (anakrish) [reply debugger/oe-gdb]: Yes. It is good that it cannot be executed.

File: cmake/openenclave-config.cmake.in [hunk 5]
```diff
@@ -22,6 +22,10 @@ find_dependency(Threads)
 add_executable(openenclave::oeedger8r IMPORTED)
 set_target_properties(openenclave::oeedger8r PROPERTIES IMPORTED_LOCATION ${OE_BINDIR}/oeedger8r)
 
+# Similarly, this is a shell script.
+add_executable(openenclave::oegdb IMPORTED)
+set_target_properties(openenclave::oegdb PROPERTIES IMPORTED_LOCATION ${OE_BINDIR}/oegdb)
```
Comment (anakrish) [inline cmake/openenclave-config.cmake.in]: @andschwa
It is not clear what openenclave::oegdb target provides the user. There ought to be some documentation or sample that demonstrates the usecase.

Comment (andschwa) [reply cmake/openenclave-config.cmake.in]: It isn't? It's just an executable 😆 I'll add to the docs.

**Pull request merged**